lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -467,36 +498,95 @@ GetNominal(swift::Demangle::Demangler &dem, swift::Demangle::NodePointer node) {
     auto *type = node->getChild(0);
     if (!type || type->getKind() != Node::Kind::Type || !type->hasChildren())
       return {};
-    return GetNominal(dem, type->getFirstChild());
+    return BuildDeclContext(dem, type->getFirstChild(), context);
   }
-  default:
-    break;
+  case Node::Kind::Function: {
+    if (node->getNumChildren() != 3)
+      return false;
+
+    auto *first = node->getChild(0);
+    auto *second = node->getChild(1);
```

Comment:
same thing here about factoring this out. I think this will get a lot more readable if it starts with:
```
std::optional<std::pair<StringRef, StringRef>> module_ident = extractFunctionNode(node);
```

Reply:
Like, the other comment, there's not much we can assume about the children, so the helper couldn't do much besides verifying the number of children and just returning them.